zxlive/eitem.py

```diff
@@ -197,3 +211,70 @@ def compute_perpendicular_direction(source_pos: QPointF, target_pos: QPointF) ->
     direction = direction / norm
     perpendicular = QPointF(-direction.y(), direction.x())
     return perpendicular
+
+
+class EItemAnimation(QVariantAnimation):
```

Comment:
This class and the `VItemAnimation` class shares most of the code. It will be cleaner to factor out the common code in a parent class.

Reply:
While that's true, quite a lot of it actually can't be made common without doing some generic typing that would be harder to interpret as well. This is because the EItem and VItem don't have useful common ancestor and the properties we can animate are different for each of these.

Reply:
that's fine, we can leave it to future work